### PR TITLE
Rework jobs `CI is happy` to be always run

### PR DESCRIPTION
**Pull request by FirelightFlagboy**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,7 +35,22 @@ jobs:
       - test-web-app
       - test-wasm
     runs-on: ubuntu-latest
+    if: always()
     steps:
+      - name: Debug needs
```
Comment (touilleMan) [inline .github/workflows/ci.yml:40]: What's that ?

(don't we risk of leaking sensitive env vars with a bad configuration on this ?)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,7 +35,22 @@ jobs:
...
+        env:
+          NEEDS: ${{ toJSON(needs) }}
+        run: printenv NEEDS
+
+      - run: |
+          echo "Oh No, we have jobs that have failed/cancelled/skipped :("
+          exit 42
+        if: >-
+          contains(needs.*.result, 'failure')
+          || contains(needs.*.result, 'skipped')
```
Comment (touilleMan) [inline .github/workflows/ci.yml:50]: When we don't modify rust, the rust job is still executed (but does nothing). So this check on skipped is not affected, right ?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (FirelightFlagboy) [reply .github/workflows/ci.yml:40]: It's a task that print the `needs` context more info [here](https://docs.github.com/en/actions/learn-github-actions/contexts#needs-context).

tl,dr: The needs context contains only the final state a jobs (if it fail or not) it's output (actually none of our jobs are configuring outputs variable)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (FirelightFlagboy) [reply .github/workflows/ci.yml:50]: Yes, you're right. The jobs is always executed but will performed a greatly reduced amount of work

File: .github/workflows/ci.yml (hunk 1 above)
Comment (FirelightFlagboy) [reply .github/workflows/ci.yml]: done

**Pull request merged**